Makefile

```diff
@@ -5,14 +5,14 @@ PYTEST_OPTIONS=--verbose
 help:
 	@echo "Please use \`make <target>' where <target> is one of:"
 	@echo "  help              to show this message"
-	@echo "  all               to execute test-coverage and lint"
+	@echo "  all               to execute test-coverage and pre-commit and docs stuff"
 	@echo "  clean			   to clean all pycache files"
 	@echo "  docs-clean        to remove documentation"
 	@echo "  docs-html         to generate HTML documentation"
 	@echo "  install           to install in editable mode"
 	@echo "  install-dev       to install in editable mode plus the "
 	@echo "                    dev packages"
-	@echo "  lint              to run flake8"
+	@echo "  pre-commit        to run pre-commit against all the files"
```

Comment:
Add make target for `pre-commit install --install-hooks`? Could be included in `install-dev`, if we agree `install-dev` really means "set up development environment for me".

Personally, I strongly prefer to gate `git push` on basic code quality checks. It's simply faster than commiting, pushing, waiting for CI to run and only then learning that black changes single quotes to double quotes or something.